avm/ptn/network/hub-networking/main.bicep

```
@@ -0,0 +1,512 @@
+metadata name = 'Hub Networking'
+metadata description = 'This module is designed to simplify the creation of multi-region hub networks in Azure. It will create a number of virtual networks and subnets, and optionally peer them together in a mesh topology with routing.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+//
+// Add your parameters here
+//
+
+@description('Optional. A map of the hub virtual networks to create.')
+param hubVirtualNetworks hubVirtualNetworkObject
```

I'd recommend nesting the object as you can see in my comment below. Here you can then change it to

```suggestion
param hubVirtualNetworks hubVirtualNetworkType
```